plugin.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/drone/drone-go/drone"
+	"golang.org/x/oauth2"
+)
+
+type Plugin struct {
+	DroneRepoNamespace string
+	DroneRepoName      string
+	DroneBranch        string
+	DroneBuildNumber   int64
+	GoclientOauthToken string
+	DroneSystemHost    string
+	DroneBuildEvent    string
+}
+
+func (p *Plugin) Exec() error {
+	namespace := p.DroneRepoNamespace
+	repository := p.DroneRepoName
+
+	client := getDroneClient(p.GoclientOauthToken, p.DroneSystemHost)
+	buildList := getBuildList(client, namespace, repository)
+
+	for _, build := range buildList {
+		if matchBuild(build, p.DroneBranch, p.DroneBuildNumber, p.DroneBuildEvent) {
+			log.Println(fmt.Sprintf("stoping build %d", build.Number))
+			cancelParentBuild(client, namespace, repository, build.Number)
+		}
+	}
+
+	return nil
+}
+
+func getDroneClient(token string, host string) drone.Client {
+	config := new(oauth2.Config)
+	auth := config.Client(
+		oauth2.NoContext,
+		&oauth2.Token{
+			AccessToken: token,
+		},
+	)
+	client := drone.NewClient(fmt.Sprintf("https://%s", host), auth)
+
+	return client
+}
+
+func getBuildList(client drone.Client, namespace string, repository string) []*drone.Build {
+	b, err := client.BuildList(namespace, repository, drone.ListOptions{
+		Page: 0,
+		Size: 0,
+	})
+
+	if err != nil {
+		log.Panic(err)
+	}
+
+	return b
+}
+
+func matchBuild(build *drone.Build, branch string, currentBuildNumber int64, droneBuildEvent string) bool {
+	return build.Status == "running" && build.Source == branch && build.Number != currentBuildNumber && build.Event == droneBuildEvent
```

Comment:
Há uma constante com esse valor.

```suggestion
	return build.Status == drone.StatusRunning && build.Source == branch && build.Number != currentBuildNumber && build.Event == droneBuildEvent
```

Reply:
bom demais, seu lindo

Reply:
vou por <3

Reply:
dá pra colocar o `go fmt` no ci do plugin tb ;)

Reply:
onde? nao entendi